pkg/kubelet/kubelet_pods.go

```diff
@@ -1208,3 +1243,42 @@ func (kl *Kubelet) PortForward(podFullName string, podUID types.UID, port uint16
 	}
 	return kl.runner.PortForward(&pod, port, stream)
 }
+
+// cleanupOrphanedPodCgroups removes the Cgroups of pods that should not be
+// running and whose volumes have been cleanedup.
+func (kl *Kubelet) cleanupOrphanedPodCgroups(
+	pods []*api.Pod, runningPods []*kubecontainer.Pod) error {
+	// Add all running and existing terminated pods to a set allPods
+	allPods := sets.NewString()
+	for _, pod := range pods {
+		allPods.Insert(string(pod.UID))
+	}
+	for _, pod := range runningPods {
+		allPods.Insert(string(pod.ID))
+	}
+
+	pcm := kl.containerManager.NewPodContainerManager()
+	found, err := pcm.GetAllPodsFromCgroups()
```

Comment:
seems like we want to at least do the pod checks from our caches after our inspection of the cgroup tree..